File: src/3168.minimum-number-of-chairs-in-a-waiting-room.3426/solution.ts
function minimumChairs (s: string): number {
    let seat = 0;
    let result = 0;
    for (let i = 0; i < s.length; i++) {
        if (s[i] === 'E') {
            seat--;
            result = Math.max(result, -seat);
        } else {
            seat++;
        }
    }
    return result;
}

File: src/3168.minimum-number-of-chairs-in-a-waiting-room.3426/solution.java
class Solution {
    public int minimumChairs(String s) {
        int count = 0;
        int result = 0;
        char[] chars = s.toCharArray();
        for(char c : chars){
            if(c == 'E'){
                count--;
                result = Math.min(result,count);
            }else{
                count++;
            }
        }
        return -result;
    }
}